server/src/main/java/org/elasticsearch/action/admin/indices/create/TransportCreateIndexAction.java

```diff
@@ -63,13 +73,48 @@ protected void masterOperation(Task task, final CreateIndexRequest request, fina
         }
 
         final String indexName = indexNameExpressionResolver.resolveDateMathExpression(request.index());
+
+        String mappings = request.mappings();
+        Settings settings = request.settings();
+        Set<Alias> aliases = request.aliases();
+
+        String concreteIndexName = indexName;
+        boolean isSystemIndex = false;
+
+        SystemIndexDescriptor descriptor = systemIndices.findMatchingDescriptor(indexName);
+
+        if (descriptor != null && descriptor.isAutomaticallyManaged()) {
+            isSystemIndex = true;
+            // System indices define their own settings and mappings, which cannot be overridden.
+            mappings = descriptor.getMappings();
+            settings = descriptor.getSettings();
+
+            if (descriptor.getAliasName() == null) {
+                aliases = Set.of();
+            } else {
+                concreteIndexName = descriptor.getIndexPattern();
```

Comment:
I think we might need a method to get the current concrete index name? Maybe someone defines a pattern with a * to cover a variety of system indices but we should know what the concrete value to create is (ie security_6 vs security_7).

Reply:
@jaymode @gwbrown Right now I feel that there's a conflict between defining a system index pattern with a wildcard, and being able to automatically create it, essentially because we have to define the concrete / underlying index name somewhere.

At the moment, this branch assumes that if `descriptor.isAutomaticallyManaged()` returns `true`, then `descriptor.getIndexPattern()` can be used as the concrete index name. Partly for that reason, the `Security` plugin doesn't define a descriptor for `.security-*`, but instead defines a `.security-6` descriptor and a `.security-7` descriptor, but only the latter has the `.security` alias defined.

That being said, I'm not sure how this scheme would work if we needed to introduce a `.security-8` for some reason. Clearly in a new cluster, `.security-8` should be auto-created with the `.security` alias pointing to it, but what about in an upgraded deployment?

I wonder whether a descriptor should be able to define:

   * An index pattern
   * One or more concrete index names, with optional mappings and settings
   * An alias-to-index mapping, which also represents what the latest concrete index name is
   * Perhaps a method to migrate between indices, e.g. convert `.foo-7` to `.foo-8`, and re-point the alias?

So the for Security, we'd have one descriptor containing:

   * An index pattern of `.security*`
   * An index definition for `.security-7` (it wouldn't be necessary to define one for `.security-6` any more)
   * The alias `.security`, pointing at `.security-7`.

What do you think?

Reply:
I think we should be able to separately have a reserved pattern and also have the ability to define concrete indices with aliases, so I agree with your proposal. 

There will be follow-on work involved since we should prevent creation of aliases that match the index pattern, unless the alias definition matches that which the system knows about.